### PR TITLE
Deprecate `project_version` and add `add_ons` to `ProjectMetadata`

### LINKED ISSUE #3084: Remove deprecated `project_version`
`project_version` was deprecated in favour of `kedro_init_version` in https://github.com/kedro-org/kedro/issues/2118 and https://github.com/kedro-org/kedro/pull/2219, but hasn't been removed yet.

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
Fix #3084, and related to #3188

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 * Renamed the `data_sets()` method in `Pipeline` and all references to it to `datasets()`.
 * Renamed the `create_default_data_set()` method in the `Runner` to `create_default_dataset()`.
 * Renamed all other uses of `data_set` and `data_sets` in the codebase to `dataset` and `datasets` respectively.
+* Deprecated `project_version` from `ProjectMetadata`.
```
Comment (merelcht) [inline RELEASE.md:15]:
```suggestion
* Remove deprecated `project_version` from `ProjectMetadata`.
```

**Review by SajidAlamQB — APPROVED** [review]
👍

**Review by ankatiyar — APPROVED** [review]
Thanks @noklam!

**Review by merelcht — APPROVED** [review]
LGTM 👍

**Pull request merged**